src/peak_finding/1d_list.clj

```diff
@@ -0,0 +1,54 @@
+;; A peak is any element that is larger or equal to both it's neighbours
+;; Items that only have one neighbour (end items) are considered a peak if
+;; they are larger or equal to their neighbour
+
+(ns peakfinder
+  (:require [clojure.core.match :refer [match]]))
+
+(defun find-peak [input]
```

Comment:
I think it would be better if this function threw an exception on an invalid input instead of returning nil.